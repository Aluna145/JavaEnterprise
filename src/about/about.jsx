import React from 'react'
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';

import Typography from '@material-ui/core/Typography';

const styles = {

	gridBoard: {
		borderRadius: '3px',
		minHeight: '82vh',
		marginLeft: 16,
		marginRight: 16,
		marginBottom: 16,
		marginTop: 20,
		background: 'white'
	},
	gridBack: {
		// objectFit: 'cover',
		backgroundSize: 'cover',
		// filter: blur(100),
		marginBottom: 10,
		// background: 'linear-gradient(to top, #66ccff 0%, #ff99cc 100%)'
		backgroundImage: `url('https://www.vactualpapers.com/wallpapers/a-beautiful-sunset-by-the-beach-nature-landscape-4k-wallpaper/download/5120x2880.jpg')`,

		// background: 'linear-gradient(to top, #66ccff 0%, #ff99cc 100%)'
	},
};
export const About = (props) => (
	<Grid container spacing={0}
	      alignItems="center"
	      justify="center" style={styles.gridBack}
	>
		<Grid container spacing={24}
		      alignItems="center"
		      justify="center" style={styles.gridBoard}
			// style={{ minHeight: '100vh' }}
		>
			<Grid>
				<Typography component="h2" variant="h1" gutterBottom>
					TIBO
				</Typography>
				<Typography variant="h3" gutterBottom>
					Travel Inspiration BOard
				</Typography>
				<Typography variant="h3" gutterBottom>
					Web-application for
				</Typography>
				<Typography variant="h3" gutterBottom>
					Here you can add (follow) cities that you plan to visit,
					see and compare information from different cities to decide your next trip. Not enougth inspiration
					to visit this or that city? Go and see pictures.
				</Typography>
				<Typography variant="h3" gutterBottom>
					For now you can add (follow) city card and see next information:
					picture with views of city
					country
					current weather
					currency
					official language
					current date and time
					approximate flight cost from your location to desired city and dates for this cost
					Also, if you don't like city anymore, you can unfollow (delete from your board)

					For future you will
					see weather monthly graph to plan your farther trips
					see more city pictures, life videos
					see upcoming events
					see unsafe places (which will be highlighted in red), recommended places (which will be highlighted in blue; "recommendedness" based on system data: cost, weather, city traffic, etc.)
					compose your travel history (we plan to add worldwide map)
					choose date "TO FROM" yourself
					filter ticket cost: "the cheapest flight ticket", "flight ticket with luggage"
					change design themes
					for the distant future you will:
					add the information you consider necessary to the city cards; Perhaps your card will be the most
					informative and will be liked by other users.
				</Typography>

			</Grid>
		</Grid>
	</Grid>

);

About.propTypes = {
	title: PropTypes.string.isRequired,
	children: PropTypes.node,
	news: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
	theme: PropTypes.oneOf(['dark', 'light'])
};

About.defaultProps = {
	title: 'News app'
};