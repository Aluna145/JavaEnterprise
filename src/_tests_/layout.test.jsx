import React from 'react'
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom'
import createStore from 'redux-mock-store'
import {Provider} from 'react-redux'
import TestRenderer from'react-test-renderer';
import renderer from 'react-test-renderer'

import {About} from '../about/about'

describe('testing Layout (toolbar) component', () => {

	it('test render layout component', async () => {
		// const initialState = {
		// 	news: []
		// };
		// const mockStore = createStore([]);
		// const store = mockStore(initialState);

		// const container = await renderer
		// 	.create(<Provider store={store}>
		// 			<Router>
		// 				<About />
		// 			</Router>
		// 		</Provider>
		// 		, {
		// 			createNodeMock: ({type}) => document.createElement(type)
		// 		});
		const instance=TestRenderer.create(
			<About/>
		);

		// const testInstance = container.getInstance();
		// testInstance.someMethod();
		// expect(container.toJSON()).toMatchSnapshot();
		expect(instance.toJSON()).toMatchSnapshot();
	});
});