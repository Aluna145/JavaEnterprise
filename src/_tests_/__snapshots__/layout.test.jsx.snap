// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Layout (toolbar) component test render layout component 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
  style={
    Object {
      "backgroundImage": "url('https://www.vactualpapers.com/wallpapers/a-beautiful-sunset-by-the-beach-nature-landscape-4k-wallpaper/download/5120x2880.jpg')",
      "backgroundSize": "cover",
      "marginBottom": 10,
    }
  }
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
    style={
      Object {
        "background": "white",
        "borderRadius": "3px",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 20,
        "minHeight": "82vh",
        "padding": 20,
      }
    }
  >
    <div
      className=""
    >
      <h2
        className="MuiTypography-root-98 MuiTypography-h1-110 MuiTypography-gutterBottom-125"
      >
        TIBO
      </h2>
      <h3
        className="MuiTypography-root-98 MuiTypography-h3-112 MuiTypography-gutterBottom-125"
      >
        Travel Inspiration BOard
      </h3>
      <h6
        className="MuiTypography-root-98 MuiTypography-h6-115 MuiTypography-gutterBottom-125"
      >
        Here you can add (follow) cities that you plan to visit, see and compare information from different cities to decide your next trip. Not enought inspiration to visit this or that city? Go and see pictures.
      </h6>
      <h3
        className="MuiTypography-root-98 MuiTypography-h3-112 MuiTypography-gutterBottom-125"
      >
        What you can do now?
      </h3>
      <h5
        className="MuiTypography-root-98 MuiTypography-h5-114 MuiTypography-gutterBottom-125"
      >
        For now you can add (follow) city card and see next information:
      </h5>
      <h6
        className="MuiTypography-root-98 MuiTypography-h6-115 MuiTypography-gutterBottom-125"
      >
        •   picture with views of city
        <br />
        •   country
        <br />
        •   monthly weather
        <br />
        •   current weather
        <br />
        •   currency
        <br />
        •   official language
        <br />
        •   current date and time
        <br />
        •   approximate flight cost from your location to desired city and dates for this cost
        <br />
        Also, if you don't like city anymore, you can unfollow (delete from your board)
        <br />
      </h6>
      <h5
        className="MuiTypography-root-98 MuiTypography-h5-114 MuiTypography-gutterBottom-125"
      >
        For future you will
      </h5>
      <h6
        className="MuiTypography-root-98 MuiTypography-h6-115 MuiTypography-gutterBottom-125"
      >
        •   see weather monthly graph to plan your farther trips
        <br />
        •   see more city pictures, life videos
        <br />
        •   see upcoming events
        <br />
        •   see unsafe places (which will be highlighted in red), recommended places (which will be highlighted in blue; "recommendedness" based on system data: cost, weather, city traffic, etc.)
        <br />
        •   compose your travel history (we plan to add worldwide map)
        <br />
        •   choose date "TO FROM" yourself
        <br />
        •   filter ticket cost: "the cheapest flight ticket", "flight ticket with luggage"
        <br />
        •   change design themes
        <br />
        •   for the distant future you will:
        <br />
        •   add the information you consider necessary to the city cards; Perhaps your card will be the most informative and will be liked by other users.
      </h6>
    </div>
  </div>
</div>
`;
